monai/data/utils.py

```diff
@@ -762,3 +764,17 @@ def __init__(self, even_divisible: bool = True, *args, **kwargs):
             if self.rank + extra_size >= self.num_replicas:
                 self.num_samples -= 1
             self.total_size = data_len
+
+
+def default_hashing(item) -> bytes:
+    """
+
+    Args:
+        item: data item to be hashed
+
+    Returns: the corresponding hash key
+
+    """
+    # TODO: Find way to hash transforms content as part of the cache
+    cache_key = hashlib.md5(json.dumps(item, sort_keys=True).encode("utf-8")).hexdigest()
```

Comment:
Is hashing the pickle not a viable thing? We're relying on json to encode objects properly.

Reply:
thanks, I've added the pickle option, it works fine with numpy arrays, so it'll address https://github.com/Project-MONAI/MONAI/issues/921